src/main/java/pl/shopgen/controllers/Search.java

```diff
@@ -0,0 +1,49 @@
+package pl.shopgen.controllers;
+
+import javafx.util.converter.BigDecimalStringConverter;
+import org.springframework.web.bind.annotation.*;
+import pl.shopgen.models.CategoryRepository;
+import pl.shopgen.models.Product;
+import pl.shopgen.models.ProductRepository;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/search")
+public class Search {
+    private final ProductController productController;
+
+    public Search(CategoryRepository categoryRepository, ProductRepository productRepository, ProductController productController) {
+        this.productController = productController;
+    }
+
+    @RequestMapping(value = {"", " "}, method = RequestMethod.GET)
+    @ResponseBody
+    List<Product> findByName(@RequestParam Map<String,String> allRequestParams) {
+        List<Product> productList = new ArrayList<Product>();
+        String  productName = allRequestParams.getOrDefault("name", null);
+        String  categoryName =allRequestParams.getOrDefault("category",null);
+        String lowerPriceString = allRequestParams.getOrDefault("lowerPrice","0");
+        String  higherPriceString =allRequestParams.getOrDefault("higherPrice","10000000");
+        BigDecimal higherPrice=new BigDecimal(higherPriceString);
+        BigDecimal lowerPrice=new BigDecimal(lowerPriceString);
+        Predicate<Product> nameFilter = p ->   productName == null ? true :  p.getName().equalsIgnoreCase(productName);
+        Predicate<Product> categoryFilter = p ->   categoryName == null ? true :  p.getCategory().getName().equalsIgnoreCase(categoryName);
+        Predicate<Product> priceFilterLower = p ->   lowerPriceString == "0" ? true :  p.getPrice().compareTo(lowerPrice)>=1;
```

Comment:
jak już to sprawdzenie czy null
a.compareTo(b)
-1 gdy b jest większe
0 gdy są równe
1 gdy a jest większe